packages/gitbook/src/components/Header/HeaderMobileMenuButton.tsx

```diff
@@ -0,0 +1,43 @@
+'use client';
+
+import { Icon } from '@gitbook/icons';
+import { useEffect } from 'react';
+
+import { useMobileMenuSheet } from '@/components/MobileMenu/useMobileMenuSheet';
+import { tString, useLanguage } from '@/intl/client';
+import { tcls } from '@/lib/tailwind';
+import { usePathname } from 'next/navigation';
+
+/**
+ * Button to show/hide the table of content on mobile.
+ */
+export function HeaderMobileMenuButton(
+    props: Partial<React.ButtonHTMLAttributes<HTMLButtonElement>>
+) {
+    const language = useLanguage();
+    const pathname = usePathname();
+    const { open, setOpen } = useMobileMenuSheet();
+
+    const toggleNavigation = () => {
+        setOpen(!open);
+    };
+
+    // biome-ignore lint/correctness/useExhaustiveDependencies: Close the navigation when navigating to a page
+    useEffect(() => {
+        setOpen(false);
+    }, [pathname]);
```

Comment:
Why do we have this logic in the MobileMenuButton and not elsewhere? A Provider or something like that?

Reply:
You will re-render the button when the pathname change for nothing.